src/Websitetoolboxforum.php

```diff
@@ -70,109 +71,187 @@ public function init()
+    }
+    public function afterSaveSettings(): void{
+        if(isset($_POST['settings']['forumUsername'])){ 
+            $forumType  = Craft::$app->getProjectConfig()->get('plugins.websitetoolboxforum.settings.forumEmbedded',false);            
+            $forumUrl  = Craft::$app->getProjectConfig()->get('plugins.websitetoolboxforum.settings.forumUrl',false);
+
             $userName               = $_POST['settings']['forumUsername'];
             $userPassword           = $_POST['settings']['forumPassword'];
-            $postData               = array('action' => 'checkPluginLogin', 'username' => $userName,'password'=>$userPassword);
-            $result                 = $this->sso->sendApiRequest('POST',WT_SETTINGS_URL,$postData,'json');
+            
+            $postData = array('action' => 'checkPluginLogin', 'username' => $userName,'password'=>$userPassword, 'plugin' => 'craft', 'websiteBuilder' => 'craftcms');           
+            $result = $this->sso->sendApiRequest('POST',WT_SETTINGS_URL,$postData,'json');
+            //echo '<pre>';print_r($result);exit;
+            if(empty($result) || (isset($result->errorMessage) && $result->errorMessage != '')){
+                if(empty($result)){
+                    $errorMessage = 'Authentication fail for Websitetoolboxforum';
+                }else{
+                    $errorMessage = $result->errorMessage;
+                }
+                Craft::$app->getSession()->setError(Craft::t('websitetoolboxforum', $errorMessage));
+                Craft::$app->getResponse()->redirect(UrlHelper::cpUrl('settings/plugins/websitetoolboxforum'))->send();exit;
+            }
             $deleteForumUrlRows     = Craft::$app->getProjectConfig()->remove('plugins.websitetoolboxforum.settings.forumUrl');
-            $deleteForumApiKeyRows  = Craft::$app->getProjectConfig()->remove('plugins.websitetoolboxforum.settings.forumApiKey');      
+            $deleteForumApiKeyRows  = Craft::$app->getProjectConfig()->remove('plugins.websitetoolboxforum.settings.forumApiKey');
+            
             $affectedForumUrlRows   = Craft::$app->getProjectConfig()->set('plugins.websitetoolboxforum.settings.forumUrl',$result->forumAddress); 
+
             $affectedForumApiKeyRows = Craft::$app->getProjectConfig()->set('plugins.websitetoolboxforum.settings.forumApiKey',$result->forumApiKey);
-        } else{
-            $userName               = Craft::$app->getProjectConfig()->get('plugins.websitetoolboxforum.settings.forumUsername',false);;
-            $userPassword           = Craft::$app->getProjectConfig()->get('plugins.websitetoolboxforum.settings.forumPassword',false);;
-            $postData               = array('action' => 'checkPluginLogin', 'username' => $userName,'password'=>$userPassword);
-            $result                 = $this->sso->sendApiRequest('POST',WT_SETTINGS_URL,$postData,'json'); 
+            if($forumUrl !='' && $forumType ==''){
```

Comment:
Please check this condition, becasue you will get forum URL in both cases while the forum is embedded or not.

Reply:
It is the case when client request to change the login setting, at that time forumUrl is set and forumType might be blank. If such a case then we need to keep the embedded url blank at admin side.